client/src/pages/Messages/MessagesDashboard.tsx

```diff
@@ -0,0 +1,250 @@
+import { useEffect, useState, useRef } from 'react';
+import {
+  Box,
+  Typography,
+  Grid,
+  List,
+  ListItem,
+  ListItemIcon,
+  Avatar,
+  ListItemText,
+  Divider,
+  Paper,
+  Button,
+  TextField,
+  Modal,
+} from '@mui/material';
+import PageContainer from '../../components/PageContainer/PageContainer';
+import { useAuth } from '../../context/useAuthContext';
+import ConversationInterface from '../../interface/Conversation';
+import getConversations from '../../helpers/APICalls/Messages/getConversations';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import moment from 'moment';
+import sendMessage from '../../helpers/APICalls/Messages/sendMessage';
+import CircularProgress from '@mui/material/CircularProgress';
+import Message from './Message/Message';
+import SendIcon from '@mui/icons-material/Send';
+import CreateConversation from './CreateConversation';
+import useStyles from './useStyles';
+
+type Conversations = ConversationInterface[];
+
+export default function MessagesDashboard(): JSX.Element {
+  const classes = useStyles();
+
+  const [conversations, setConversations] = useState<Conversations>([]);
+  const [isSubmitting, setSubmitting] = useState(false);
+  const { updateSnackBarMessage } = useSnackBar();
+  const [chatbox, setChatBox] = useState<number>(0);
+  const { loggedInUser } = useAuth();
+  const [message, setMessage] = useState<string>('');
+  const scrollRef = useRef<HTMLLIElement>(null);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
```

Comment:
```suggestion
  const handleOpen = () => setOpen((prevState) => !prevState);
```
State updates may be asynchronous: React sometimes batches multiple setState() calls for performance reasons. Let's use use a function which takes the previous state as a first argument.